File: Server/CustomerManagementProject/CustomerManagementProject/Models/Contract.cs
using System.ComponentModel.DataAnnotations;

namespace CustomerManagementProject.Models
{
    public class Contract
    {
        public int Id { get; set; }
        public string Number { get; set; }
        public string Name { get; set; }
        public ContractType ContractTypes { get; set; }
        public List<Package>? Packages { get; set; }
        public int? CustomerId { get; set; }
        //public Customer? Customer { get; set; }
    }

    public enum ContractType
    {
        Type1,
        Type2,
        Type3
    }
}

File: Client/AngularApp/src/models/Contract.ts
import { Package } from './Package';

export interface Contract {
  id: number;
  customerId: number;
  name: string;
  number: string;
  contractTypeId: number;
  contractTypes: ContractTypes;
  packages: Package[];
}

export enum ContractTypes {
  Type1,
  Type2,
  Type3,
}
